public/js/cal.js

```diff
@@ -1,269 +1,269 @@
-  selectedDay = displayedDate;
-}
-
-function changeView() {
-  document.getElementById("popup").className = "hidden";
-  if (currentView === "week") {
-    document.getElementById("weekView").classList.toggle("hidden");
-    document.getElementById("dayView").classList.toggle("hidden");
-    currentView = "day";
-  } else if (currentView === "day") {
-    document.getElementById("weekView").classList.toggle("hidden");
-    document.getElementById("dayView").classList.toggle("hidden");
-    currentView = "week";
-  }
-  while (document.getElementsByClassName("selected")[0]) {
-    document.getElementsByClassName("selected")[0].className = "";
-  };
-  updateGrid();
-}
-
-socket.on("calendar event", function(event){
-  event.time = new Date(new Date(event.time).setHours(new Date(event.time).getHours())).toLocaleString();
-  events.push(event);
-  document.getElementById("popup").className = "hidden";
-  while (document.getElementsByClassName("selected").length > 0) {
-    document.getElementsByClassName("selected")[0].classList.remove("selected");
-  }
-  updateGrid();
-});
-
-socket.on("delete calendar event", function(event) {
-  for (var i = 0; i < events.length; i++) {
-    if (new Date(events[i].time).toLocaleString() === event.time && events[i].room === event.room) {
-       events.splice(i,1);
-    }
-  }
-  document.getElementById("popup").className = "hidden";
-  while (document.getElementsByClassName("selected").length > 0) {
-    document.getElementsByClassName("selected")[0].classList.remove("selected");
-  }
-  updateGrid();
-});
-
-function deleteCell(event) {
-  if(confirm("Are you sure you want to delete this room reservation?")){
-    cell = event.srcElement;
-    eventDate = new Date((currentView === "week") ? new Date(new Date().setDate(window.now.getDate()+7*weekDiff+(cell.dataset.day-window.now.getDay()))) : cellDate = new Date(displayedDate)).toDateString();
-    eventTime = new Date(new Date(eventDate).setHours(cell.dataset.time)).toLocaleString();
-    socket.emit('delete calendar event', {"room":(currentView === "week" ? document.getElementById("displayRoomSelect").value : cell.dataset.room), "time":eventTime, "user":window.userName});
-  }
-}
-
-function submit() {
-  eventTitle = document.getElementById("nameInput").value;
-  eventTime = new Date(document.getElementById("dateSelect").value+" "+Number(document.getElementById("timeSelect").value)+":00:00").toLocaleString();
-  selectedRoom = document.getElementById("roomSelect").value;
-  selectedDuration = document.getElementById("durationSelect").value;
-  if (document.getElementById("userInput")) {
-    var user = document.getElementById("userInput").value;
-  } else {
-    var user = window.userName;
-  }
-  
-  socket.emit('calendar event', {"user":user, "time":eventTime, "room":selectedRoom, "duration":selectedDuration, "title":eventTitle});
-}
-})();
+(function(){
+
```

Comment:
No wonder your graph looks so good [how does it feel?](https://github.com/BYU-ODH/hlrdesk/graphs/contributors)